### PR TITLE
Add paper for scikit-optimize

### PR DESCRIPTION
**Pull request by MechCoder**
Fixes https://github.com/scikit-optimize/scikit-optimize/issues/263

### CODE REVIEW

**Comment by MechCoder** [comment]
This is a short-term solution. I feel we should submit to JMLR OSS after some-time.

1. Did I forget to cite any paper?
2. There are more names to be added. Do we add all of the 16 contributors?

**Comment by betatim** [comment]
Can we make the author "The scikit-optimize contributors" with a link to the actual list? Inspired by what the big collaborations at CERN do. It could be a nice way to have the list of authors "update" as time passes. Thoughts?

**Comment by MechCoder** [comment]
can you give me an example?

**Comment by betatim** [comment]
https://arxiv.org/abs/1207.7214 this is what I was thinking. The only author name you see on the arxiv page is "The ATLAS collaboration". Maybe we can ask the JOSS guys for a recommendation as part of the submission (I am sure it will generate a bit of discussion).

**Comment by MechCoder** [comment]
ok I'll add all the contributors for now and then we can comment on the `joss-reviews` thread.

**Comment by MechCoder** [comment]
Btw, you are a member here (https://github.com/openjournals). So you are one of the "JOSS guys"

**Comment by MechCoder** [comment]
ping @glouppe @betatim @cmmalone @nel215 @nfcampos @mirca @thomasjpfan @mp4096 @iaroslav-ai @Naereen @lefant @jtsmith2 @andreh7 @mehdidc @AlexanderFabisch 

Can you please verify your names? Can you also let me if you want to add your affliations and ORCID ids?

I'll leave it open till next week.

**Comment by betatim** [comment]
> Btw, you are a member here (https://github.com/openjournals). So you are one of the "JOSS guys"

Yes, I reviewed the [first paper ever](https://github.com/openjournals/joss-reviews/issues/11) ... this might explain why I like JOSS.

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  - Python
+  - Machine-learning
+  - Global optimization
+  - Scientific Computing
+authors:
+ - name: Manoj Kumar
+   orcid: 0000-0002-3802-8346
+   affiliation: 1
+ - name: Tim Head
+   orcid:
```
Comment (betatim) [inline paper/paper.md]: 0000-0003-0931-3698

My affiliation is "Wild Tree Tech"

**Comment by glouppe** [comment]
Regarding authors, i would go for something like "The Scikit-Optimizer development team: A. Foo, B. Bar, ...".

See e.g. https://arxiv.org/abs/1605.02688

**Comment by MechCoder** [comment]
How do I format the table in such a way?

**Comment by glouppe** [comment]
Also, what do you want to tell in this paper? While I like the motives behind JOSS, I have been quite surprised by the actual content of those 'papers'. Most of them seem to be empty shells stamped with a DOI, and not much more than that. I find submissions to JMLR OSS to usually be of higher quality.

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+   affiliation: 2
+ - name: Gilles Louppe
```
Comment (glouppe) [inline paper/paper.md]: 000-0002-2082-3106
NYU

**Comment by glouppe** [comment]
Other question: if we eventually want to make a submission to JMLR OSS, which paper should be cited?

File: paper/paper.bib [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+         and Weiss, R. and Dubourg, V. and Vanderplas, J. and Passos, A. and
+         Cournapeau, D. and Brucher, M. and Perrot, M. and Duchesnay, E.},
+ journal={Journal of Machine Learning Research},
+ volume={12},
+ pages={2825--2830},
+ year={2011}
+}
+
+@article{jones2014scipy,
+  title={$\{$SciPy$\}$: open source scientific tools for $\{$Python$\}$},
```
Comment (Naereen) [inline paper/paper.bib]: Why the `$\{$` ? Isn't `{{SciPy} ... {Python}}` enough?

**Comment by Naereen** [comment]
@MechCoder That's an interesting initiative. Nice to ping all the contributors.
I share some of @glouppe's doubts, but no matter how and where it is published it will be useful.

- Also, if you "just want to have a DOI so people cite you", [Zenodo](https://zenodo.org/) and [its GitHub integration](https://guides.github.com/activities/citable-code/) (+ publishing a notice on Google Scholar) might be enough...
- The paper starts to look good. How can we help? Using PR if everyone starts writing will be a mess, maybe you can do a first draft and we review it (one by one) ?

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+   affliation: 8
+ - name: Mikhail Pak
+   orcid:
+   affliation: 9
+ - name: Iaroslav Shcherbatyi
+   orcid:
+   affliation: 10
+ - name: Lilian Besson
+   orcid:
+   affliation: 11
```
Comment (Naereen) [inline paper/paper.md]: - orcid : [0000-0003-2767-2563](https://orcid.org/0000-0003-2767-2563)

File: paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+ - name:
+   index: 7
+ - name:
+   index: 8
+ - name:
+   index: 9
+ - name:
+   index: 10
+ - name:
+   index: 11
```
Comment (Naereen) [inline paper/paper.md]: Affiliation `11` for @Naereen: [CentraleSupélec](http://orcid.org/0000-0002-6972-589X)
(if the accent is messing up, you can use `{\'e}` if the `paper.md` file aims at being compiled to PDF through pandoc and LaTeX)

File: paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+ - name: Lilian Besson
+   orcid:
+   affliation: 11
+ - name: Fabian Linzberger
+   orcid:
+   affliation: 12
+ - name: Taylor Smith
+   orcid:
+   affliation: 13
+ - name: andreh7
```
Comment (andreh7) [inline paper/paper.md]: Andre Holzner

File: paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+   orcid:
```
Comment (andreh7) [inline paper/paper.md]: orcid: [0000-0003-0863-3825](http://orcid.org/0000-0003-0863-3825)

File: paper/paper.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+   index: 10
+ - name:
+   index: 11
+ - name:
+   index: 12
+ - name:
+   index: 13
+ - name:
+   index: 14
+ - name:
```
Comment (andreh7) [inline paper/paper.md]: University of California,  San Diego,  La Jolla,  USA

**Comment by andreh7** [comment]
I'm not sure my little contribution to scikit-optimize deserves
co-authorship but of course I'd be very happy to contribute to such a paper
e.g. with editing work. I have added my real name, ORCID and affiliation as
comments to `paper.md`.

**Comment by Naereen** [comment]
Same feeling as @andreh7 https://github.com/scikit-optimize/scikit-optimize/pull/387#issuecomment-305419009

File: paper/paper.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+ - name: Nuno Campos
+   orcid:
+   affliation: 6
+ - name: Zé Vinícius
+   orcid:
+   affliation: 7
+ - name: Thomas Fan
+   orcid:
+   affliation: 8
+ - name: Mikhail Pak
```
Comment (mp4096) [inline paper/paper.md]: Mikhail Pak

orcid: 0000-0002-6852-9475

affiliation: Technical University of Munich, Munich, Germany

File: paper/paper.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+   affiliation: 2
+ - name: Gilles Louppe
+   orcid:
+   affiliation: 3
+ - name: Katie Malone
+   orcid:
+   affiliation: 4
+ - name: nel215
+   orcid:
+   affliation: 5
```
Comment (mp4096) [inline paper/paper.md]: From this line one there's a typo: `aff_i_liation` :upside_down_face:

**Comment by mp4096** [comment]
Same feeling as @andreh7 and @Naereen :smiley:

**Comment by betatim** [comment]
I think we should create something that people can cite, "just a DOI". This gives people an easy way to say "we used scikit-optimize". Undecided if it is better to create on DOI per release (we sum over all DOIs if we want to know who uses it) or just have one for all versions (authors mention which version they use "...as implemented in scikit-optimize v0.6 [3]." For this IMHO we don't need much text in the paper. It is a signpost. So maybe using just zenodo is enough.

Separate from that for me is writing a paper on new ideas or smart implementations. This should be a paper that contains text to explain the thinking etc. Writing such a paper would be a much bigger exercise and we can't do it all the time (because we don't have that many ideas nor that much time ;))

**Comment by mirca** [comment]
Same feeling as @andreh7, @Naereen, @mp4096 :)

**Comment by AlexanderFabisch** [comment]
@MechCoder I think you can remove me from the author list. I contributed exactly one line. :)

File: paper/paper.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+   affliation: 7
+ - name: Thomas Fan
+   orcid:
+   affliation: 8
+ - name: Mikhail Pak
+   orcid:
+   affliation: 9
+ - name: Iaroslav Shcherbatyi
+   orcid:
+   affliation: 10
```
Comment (iaroslav-ai) [inline paper/paper.md]: My affiliation is "Saarland University"

File: paper/paper.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+ - name: Fabian Linzberger
+   orcid:
+   affliation: 12
+ - name: Taylor Smith
+   orcid:
+   affliation: 13
+ - name: andreh7
+   orcid:
+   affliation: 14
+ - name: Mehdi Cherti
```
Comment (mehdidc) [inline paper/paper.md]: ORCID : 0000-0001-5865-0469
My affiliation:  
Center of Data Science, Université Paris Saclay
LAL, CNRS, France

File: paper/paper.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+   orcid:
+   affliation: 15
```
Comment (mehdidc) [inline paper/paper.md]: ORCID : 0000-0001-5865-0469
My affiliation:
Center of Data Science, Université Paris Saclay
LAL, CNRS, France

File: paper/paper.md [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+ - name: Tim Head
+   orcid:
+   affiliation: 2
+ - name: Gilles Louppe
+   orcid:
+   affiliation: 3
+ - name: Katie Malone
+   orcid:
+   affiliation: 4
+ - name: nel215
```
Comment (nel215) [inline paper/paper.md]: name: Yuhei Otomo
orcid: 0000-0001-7198-1449

File: paper/paper.md [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+
+affiliations:
+ - name: New York University
+   index: 1
+ - name:
+   index: 2
+ - name:
+   index: 3
+ - name:
+   index: 4
```
Comment (cmmalone) [inline paper/paper.md]: affiliation 4 = Civis Analytics

File: paper/paper.md [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+# Summary
+scikit-optimize is a sequential model-based optimization toolkit.
+
+scikit-optimize aims at the global minimization of functions that are noisy,
+expensive or for which gradients are not available. Examples of such optimization
+includes optimizing the loss of a deep neural network on validation-data or
+the settings of a robot controller. The technique used by
+scikit-optimize is popularly known as sequential optimization or
+bayesian optimization. The key idea behind this technique is
+that a predictive model is used to suggest a new candidate point given the
```
Comment (iaroslav-ai) [inline paper/paper.md]: Here one might understand this as if predictive model is used whose output is the next point which will be evaluated. 

Consider using something along the lines: "At every iteration, predictive model is trained on the set of points which has been already evaluated to estimate the value of objective given some new point. A special class of models is used, which can give an 'exploration' score, which indicates how close the point is to already evaluated points (in some sense). Then the next point for evaluation is such that it minimizes a tradeoff between the estimated objective value and 'exploration' score. "

File: paper/paper.md [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+function evaluations at the previous points. This new point is suggested
+such that there is a balance between exploitation and exploration.
+
+The aim of scikit-optimize is to provide an easy to use interface for such
```
Comment (iaroslav-ai) [inline paper/paper.md]: Here one could understand this as if scikit-optimize is just simplifying the interface of other software. Maybe something like this:
"The aim of scikit-optimize is to provide an implementation of such optimization techniques with easy to use interface."

File: paper/paper.md [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+optimization techniques. Currently, we support Gaussian Process, Random Forest
+and Gradient Boosting based search and also allow the power-user to plug in a custom predictive model. In terms of
+interface, we support two interfaces, an interface that
```
Comment (iaroslav-ai) [inline paper/paper.md]: Remove the "In terms of interface" - it seems redundant, maybe "We provide two different interfaces, one that mimics ..."

File: paper/paper.md [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+ - name:
+   index: 16
+date: June 2, 2017
+bibliography: paper.bib
+---
+
+# Summary
+scikit-optimize is a sequential model-based optimization toolkit.
+
+scikit-optimize aims at the global minimization of functions that are noisy,
```
Comment (betatim) [inline paper/paper.md]: I suggest:

scikit-optimize implements several methods for finding the approximate minimum of objective functions that fulfill the following three criteria:
* (very) expensive to evaluate;
* gradients can not be computed analytically; and
* evaluations are noisy.

Examples of such optimization problems include: tuning hyper-parameters of deep neural networks [cite a deep learning example], finding optimal parameters for a robot's control system [citation needed], or locating the best location at which to build a new oil well [cite].

**Comment by glouppe** [comment]
Before going further, we should decide on what we do with this.

Options include:
1) Making a Zenodo submission for the software (through which we will get a DOI for the actual code, not for an attached paper). Note that Zenodo now supports versions. 
2) Writing an abstract to be submitted to JOSS, as proposed here (through which we will get a DOI for the abstract).
3) Writing a paper with some actual content and submit that to arXiv+JMLR-OSS. 

As far as I am concerned, I am more in favour of doing 1) and then 3) (as commonly done in ML software). 

I am not against 2), but I genuinely do not yet understand how it is better than Zenodo? Is it in the fact that JOSS is peer-reviewed?

**Comment by MechCoder** [comment]
> I am not against 2), but I genuinely do not yet understand how it is better than Zenodo? Is it in the fact that JOSS is peer-reviewed?

That is what I thought when @betatim suggested it first, Maybe he has a stronger argument?

**Comment by Naereen** [comment]
I think we have enough smart people here to try to write a "non-empty article", so I am in favor of 3.

**Comment by betatim** [comment]
The only reason to prefer JOSS over Zenodo is that I think it comes with a bit more visibility. However if Zenodo has versions I think we should do that instead of submitting to JOSS for each version (and finding out how funny/not funny the JOSS guys find that).

The problem I would like to see solved is: "I used skopt in my work, how can I cite it?" and solve it in a way that is scalable for us to do every release (or at least more frequently than writing a 3-5 page paper).

**Comment by MechCoder** [comment]
ok then let us zenodo. Sorry for the trouble guys.

**Comment by betatim** [comment]
Closing this then or should we use it to instead write a short abstract for zenodo? Should we gather the ORCID IDs somewhere?

**Comment by betatim** [comment]
Closing for the moment.
